lib/semian/net_http.rb

```diff
@@ -0,0 +1,85 @@
+require 'semian'
+require 'semian/adapter'
+require 'net/https'
+require 'open-uri'
+
+module Net
+  ::Net::ProtocolError.include(::Semian::AdapterError)
+
+  class SemianError < ::Net::ProtocolError
+    def initialize(semian_identifier, *args)
+      super(*args)
+      @semian_identifier = semian_identifier
+    end
+  end
+
+  Net::ResourceBusyError = Class.new(SemianError)
+  Net::CircuitOpenError = Class.new(SemianError)
+end
+
+module Semian
+  module NetHTTP
+    include Semian::Adapter
+
+    ResourceBusyError = ::Net::ResourceBusyError
+    CircuitOpenError = ::Net::CircuitOpenError
+
+    DEFAULT_SEMIAN_OPTIONS = @raw_semian_options = {
+      tickets: 3,
+      success_threshold: 1,
+      error_threshold: 3,
+      error_timeout: 10,
+    }
+
+    def semian_identifier
+      "http_#{address.tr('.', '_')}_#{port}"
```

Comment:
Hmm I'm a little worried that this will create a LOT of different SysV semaphores (like, tens of thousands per day at least, in Shopify core). Is that fine? Is there no limit in the kernel for how many there can be? @Sirupsen?